### PR TITLE
Replace `test/compilable/crlf.sh` with D equivalent

### PR DESCRIPTION
**Pull request by jacob-carlborg**
Alternative implementation to #9848.

The test implementation becomes a bit messy due to the need to mix different line endings.

### CODE REVIEW

File: test/unit/compilable/crlf.d [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        /**
+          doc comment
+        */
+        void documentee() {}
+    }
+    .lineSplitter
+    .array
+    .dropBackOne
+    .dropOne
+    .map!(line => line.chompPrefix("        "))
```
Comment (jacob-carlborg) [inline test/unit/compilable/crlf.d]: These range functions here are basically a quick hack to remove the extra indentation added with by the `q{` string literal syntax.

**Review by thewilsonator — APPROVED** [review]

**Comment by jacob-carlborg** [comment]
Ideally this should probably be a test that only uses the lexer and verifies the tokens (it looks like the `static assert`s are used for this). But I think this is a good start and as close as possible to the original test.

File: test/unit/compilable/crlf.d (hunk 1 above)
Comment (marler8997) [reply test/unit/compilable/crlf.d]: You should probably replace the line endings with `\r\n` to match the original test as well.

Comment (jacob-carlborg) [reply test/unit/compilable/crlf.d]: It does, one line below: https://github.com/dlang/dmd/pull/9849/files#diff-163b1fbf9b5db86d7742341305c7b422R57

Comment (jacob-carlborg) [reply test/unit/compilable/crlf.d]: I wrote the code, both in this test and in the original test, to a file and diffed the result. No changes.

**Review by marler8997 — APPROVED** [review]
Very good stuff.

File: test/unit/compilable/crlf.d (hunk 1 above)
Comment (ZombineDev) [reply test/unit/compilable/crlf.d]: Can convert this into a function named something along the lines of `trimPrefix(string prefix, string newline = "\n")` to make it about more self-documenting?

Comment (jacob-carlborg) [reply test/unit/compilable/crlf.d]: Sure.

Comment (jacob-carlborg) [reply test/unit/compilable/crlf.d]: Fixed.

**Review by PetarKirov — APPROVED** [review]

**Comment by thewilsonator** [comment]
Is this good to go?

**Comment by jacob-carlborg** [comment]
Yes, this is good to go.

**Pull request merged**